<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="A complete workflow to perform single-omics analysis prior to integrating multi-omics data for an integrated analysis. ">
<title>A Multi-Omics Integration Toolkit • Omix</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="A Multi-Omics Integration Toolkit">
<meta property="og:description" content="A complete workflow to perform single-omics analysis prior to integrating multi-omics data for an integrated analysis. ">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">Omix</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/Omix_vignette_CV_MTG.html">&lt;b&gt;Omix&lt;/b&gt; - Integrative analysis of Multi-omics data in R</a>
    <a class="dropdown-item" href="articles/Omix_vignette_MAP_CV_pseudotemporal.html">&lt;b&gt;Omix&lt;/b&gt; - Integrative analysis of Multi-omics data in R</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><p><img src="reference/figures/Omix_logo.png" width="200" style="align:left;"></p>
<div class="section level1">
<div class="page-header"><h1 id="omix">Omix<a class="anchor" aria-label="anchor" href="#omix"></a>
</h1></div>
<!-- badges: start -->

<p>The Omix pipeline provides a generalisable framework that can effectively process, visualise and analyse patient-specific multi-omic data in an end-to-end manner. Omix is built on four consecutive blocks, summarised in the figure below.</p>
<div class="figure">
<img src="reference/figures/Omix_workflow.png" alt=""><p class="caption">Omix</p>
</div>
<p>This bioinformatics pipeline aims to address important gaps related to the integration of multi-omics data, namely by enabling the reproducibility of integrative analyses and providing grounds for more systematic benchmarking of analyses. Pipeline outputs are standardized and include publication-quality plots, tables, and interactive dashboards.</p>
<p>Currently, Omix supports the integration of bulk transcriptomics and bulk proteomics.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the development version of Omix from <a href="https://github.com/" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"eleonore-schneeg/Omix"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="overview-of-omix-features">Overview of Omix features<a class="anchor" aria-label="anchor" href="#overview-of-omix-features"></a>
</h2>
<ul>
<li>Multi-omics data container
<ul>
<li>The Omix multimodal container harmonises data management of multiple omics datasets. It enables the storage of raw and processed omics data slots, along with patient metadata, technical metadata, analysis parameters and outputs. The object structure relies on the <a href="https://bioconductor.org/packages/release/bioc/html/MultiAssayExperiment.html" class="external-link">MultiAssayExperiment library</a>
</li>
</ul>
</li>
<li>Data processing &amp; Quality Control
<ul>
<li>Each omics layer is processed separately according to best practices. Given the wide range of processing functionalities, users decide which parameters and steps of the modular sequence are performed, which involves all or a combination of the folllowing steps:</li>
<li>Feature filtering</li>
<li>Normalisation/ transformation</li>
<li>Batch correction &amp; denoising</li>
<li>Sample outlier removal</li>
<li>Formatting</li>
</ul>
</li>
<li>Single platform models
<ul>
<li>Omix provides a suite of analysis options including differential analysis (DE), a standard method to identify genes that are differentially expressed between certain disease states; <a href="https://cran.r-project.org/web/packages/WGCNA/index.html" class="external-link">Weighted Gene correlation Network (WGCNA)</a>, to identify modules of genes that associate to certain disease covariates; <a href="http://mixomics.org" class="external-link">sparse Partial Least Square (sPLS)</a> to define a sparse set of omics features, or molecular signature, that explains the response variable.</li>
</ul>
</li>
<li>Vertical integration for joint analysis
<ul>
<li>Omix provides a series of state-of-the-art integration models to perform patient-specific multi-omic integration, including:</li>
<li><a href="https://biofam.github.io/MOFA2/" class="external-link">MOFA</a></li>
<li><a href="https://biofam.github.io/MOFA2/" class="external-link">MEIFESTO</a></li>
<li><a href="http://mixomics.org" class="external-link">Sparse Multi-Block PLS (sMB-PLS)</a></li>
<li><a href="http://mixomics.org" class="external-link">DIABLO</a></li>
<li>A range of multi-omics clustering models implemented in <a href="https://github.com/xlucpu/MOVICS" class="external-link">MOVICS</a>
</li>
</ul>
</li>
<li>Downstream analyses
<ul>
<li>Multi-omics integration is followed by a series of downstream analyses, including:</li>
<li>Multi-omics networks with <a href="https://igraph.org" class="external-link">iGraph</a>
</li>
<li>Community detection with the <a href="https://igraph.org" class="external-link">Louvain or Leiden clustering algorithms</a>
</li>
<li>Pseudotime inference with <a href="https://bioconductor.org/packages/release/bioc/html/slingshot.html" class="external-link">Slingshot</a>
</li>
<li>Functional enrichment with <a href="https://maayanlab.cloud/Enrichr/" class="external-link">EnrichR</a>
</li>
<li>Cell-type enrichment with <a href="https://bioconductor.org/packages/release/bioc/html/EWCE.html" class="external-link">EWCE</a>
</li>
<li>Target validation based on the <a href="https://www.opentargets.org" class="external-link">OpenTargets</a> database</li>
<li>Publication quality plots and analysis reports</li>
</ul>
</li>
</ul>
<p>Omix implements these modular steps and displays results in an interactive results dashboard.</p>
<p>The package functions are designed to interface neatly with NextFlow for scalable and containerized pipelines deployed locally, on high-performance computing clusters, or in the cloud. See <a href="https://nf-co.re/omix" class="external-link uri">https://nf-co.re/omix</a> for the accompanying NextFlow pipeline.</p>
</div>
</div>

  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small>GPL (&gt;= 3)</small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing Omix</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Eléonore Schneegans <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-6754-6249" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Nurun Fancy <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-6481-6266" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://www.gnu.org/licenses/gpl-3.0" class="external-link"><img src="https://img.shields.io/badge/License-GPLv3-green.svg" alt="License: GPL v3"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Eléonore Schneegans, Nurun Fancy.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
