<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Function reference • Omix</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">Omix</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Omix_vignette_CV_MTG.html">&lt;b&gt;Omix&lt;/b&gt; - Integrative analysis of Multi-omics data in R</a>
    <a class="dropdown-item" href="../articles/Omix_vignette_MAP_CV_pseudotemporal.html">&lt;b&gt;Omix&lt;/b&gt; - Integrative analysis of Multi-omics data in R</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"></ul></div>

    
  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Function reference</h1>
    </div>

    <div class="section level2">
      <h2 id="all-functions">All functions<a class="anchor" aria-label="anchor" href="#all-functions"></a></h2>
      
      

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="DE_multiomics.html">DE_multiomics()</a></code> 
        </dt>
        <dd>Performs DE on multiomics object</dd>
      </dl><dl><dt>
          
          <code><a href="GSEA.html">GSEA()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="MOFA_get_relevant_factor.html">MOFA_get_relevant_factor()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="OpenTargets.html">OpenTargets()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="batch_correction_protein.html">batch_correction_protein()</a></code> 
        </dt>
        <dd>Calling different proteomics correction methods</dd>
      </dl><dl><dt>
          
          <code><a href="cell_type_enrichment.html">cell_type_enrichment()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="circular_corrplot.html">circular_corrplot()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="circular_heatmap.html">circular_heatmap()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="clustering_DE_analysis.html">clustering_DE_analysis()</a></code> 
        </dt>
        <dd>Performs DE between clusters</dd>
      </dl><dl><dt>
          
          <code><a href="combat_correction.html">combat_correction()</a></code> 
        </dt>
        <dd>Proteomics normalisation batch correction using Combat</dd>
      </dl><dl><dt>
          
          <code><a href="counts_rna.html">counts_rna()</a></code> 
        </dt>
        <dd>Adds custom processed dataset to multiassay object</dd>
      </dl><dl><dt>
          
          <code><a href="dot-communities_network.html">.communities_network()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="dot-community_graph.html">.community_graph()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="dot-get_ID_names.html">.get_ID_names()</a></code> 
        </dt>
        <dd>Convert nomenclature</dd>
      </dl><dl><dt>
          
          <code><a href="dot-get_ID_type.html">.get_ID_type()</a></code> 
        </dt>
        <dd>Get nomenclature type</dd>
      </dl><dl><dt>
          
          <code><a href="dot-get_background.html">.get_background()</a></code> 
        </dt>
        <dd>Get background genes</dd>
      </dl><dl><dt>
          
          <code><a href="dot-get_metadata.html">.get_metadata()</a></code> 
        </dt>
        <dd>Get metadata</dd>
      </dl><dl><dt>
          
          <code><a href="dot-get_multimodal_object.html">.get_multimodal_object()</a></code> 
        </dt>
        <dd>Extract the multimodal object</dd>
      </dl><dl><dt>
          
          <code><a href="dot-impute_distribution.html">.impute_distribution()</a></code> 
        </dt>
        <dd>Imputation of missing values based on distribution</dd>
      </dl><dl><dt>
          
          <code><a href="dot-impute_minimum_value.html">.impute_minimum_value()</a></code> 
        </dt>
        <dd>Imputation of missing values based on 50% minimum value</dd>
      </dl><dl><dt>
          
          <code><a href="dot-interactive_network.html">.interactive_network()</a></code> 
        </dt>
        <dd>Get interactive network from igraph object</dd>
      </dl><dl><dt>
          
          <code><a href="dot-multiomics_network.html">.multiomics_network()</a></code> 
        </dt>
        <dd>Get multiomics networks from multiassay object</dd>
      </dl><dl><dt>
          
          <code><a href="dot-multiomics_network_cluster.html">.multiomics_network_cluster()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="dot-multiomics_network_matrix.html">.multiomics_network_matrix()</a></code> 
        </dt>
        <dd>Get multiomics networks from multimodal object</dd>
      </dl><dl><dt>
          
          <code><a href="dotplot_enrichment_MOFA.html">dotplot_enrichment_MOFA()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="dotplot_enrichment_MOFA_semantics.html">dotplot_enrichment_MOFA_semantics()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="enrichment_custom.html">enrichment_custom()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="extract_multiomic_signature.html">extract_multiomic_signature()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="extract_weigths.html">extract_weigths()</a></code> 
        </dt>
        <dd>Extract_weigths</dd>
      </dl><dl><dt>
          
          <code><a href="format_res_deseq.html">format_res_deseq()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="format_res_limma.html">format_res_limma()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="generate_multiassay.html">generate_multiassay()</a></code> 
        </dt>
        <dd>Generate a MultiAssayExperiment object from single-omics data</dd>
      </dl><dl><dt>
          
          <code><a href="getMoHeatmap.html">multiomics_clustering_heatmap()</a></code> 
        </dt>
        <dd>Get multi-omics comprehensive heatmap</dd>
      </dl><dl><dt>
          
          <code><a href="get_module_annotation.html">get_module_annotation()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="integrate_with_DIABLO.html">integrate_with_DIABLO()</a></code> 
        </dt>
        <dd>Vertical integration with DIABLO including model tuning</dd>
      </dl><dl><dt>
          
          <code><a href="integrate_with_MBPLS.html">integrate_with_MBPLS()</a></code> 
        </dt>
        <dd>Vertical integration with MBPLS</dd>
      </dl><dl><dt>
          
          <code><a href="integrate_with_MEIFESTO.html">integrate_with_MEIFESTO()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="integrate_with_MOFA.html">integrate_with_MOFA()</a></code> 
        </dt>
        <dd>Vertical integration with MOFA</dd>
      </dl><dl><dt>
          
          <code><a href="integrate_with_iCluster.html">integrate_with_iCluster()</a></code> 
        </dt>
        <dd>Vertical integration with iCluster</dd>
      </dl><dl><dt>
          
          <code><a href="integrate_with_sMBPLS.html">integrate_with_sMBPLS()</a></code> 
        </dt>
        <dd>Vertical integration with SMBPLS</dd>
      </dl><dl><dt>
          
          <code><a href="integrative_results_clustering.html">integrative_results_clustering()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="integrative_results_supervised.html">integrative_results_supervised()</a></code> 
        </dt>
        <dd>Downstream analysis for integrated multi-omics data</dd>
      </dl><dl><dt>
          
          <code><a href="integrative_results_unsupervised.html">integrative_results_unsupervised()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="median_centering_correction.html">median_centering_correction()</a></code> 
        </dt>
        <dd>Proteomics normalisation batch correction using median centering method</dd>
      </dl><dl><dt>
          
          <code><a href="multiomics_modules.html">multiomics_modules()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="pathway_analysis_enrichr.html">pathway_analysis_enrichr()</a></code> 
        </dt>
        <dd>Functional enrichment analysis using enrichR</dd>
      </dl><dl><dt>
          
          <code><a href="plot_geneset_clusters.html">plot_geneset_clusters()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="process_protein.html">process_protein()</a></code> 
        </dt>
        <dd>Process bulk proteomics data</dd>
      </dl><dl><dt>
          
          <code><a href="process_rna.html">process_rna()</a></code> 
        </dt>
        <dd>Process batch transcriptomics data</dd>
      </dl><dl><dt>
          
          <code><a href="processed_proteomics.html">processed_proteomics()</a></code> 
        </dt>
        <dd>Add custom processed proteomics dataset to multiassay</dd>
      </dl><dl><dt>
          
          <code><a href="processed_rna.html">processed_rna()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="protein_DE_analysis.html">protein_DE_analysis()</a></code> 
        </dt>
        <dd>Performs single omic analysis of proteomic layer.</dd>
      </dl><dl><dt>
          
          <code><a href="pseudotime_inference.html">pseudotime_inference()</a></code> 
        </dt>
        <dd>pseudotime_inference</dd>
      </dl><dl><dt>
          
          <code><a href="pseudotime_ordering_modules.html">pseudotime_ordering_modules()</a></code> 
        </dt>
        <dd>pseudotime_ordering_modules</dd>
      </dl><dl><dt>
          
          <code><a href="rna_DE_analysis.html">rna_DE_analysis()</a></code> 
        </dt>
        <dd>Performs single omic analysis of transcriptomics layer.</dd>
      </dl><dl><dt>
          
          <code><a href="single_omic_comparison.html">single_omic_comparison()</a></code> 
        </dt>
        <dd>Compare differential expressed genes and proteins</dd>
      </dl><dl><dt>
          
          <code><a href="vertical_integration.html">vertical_integration()</a></code> 
        </dt>
        <dd>Patient specific vertical integration</dd>
      </dl><dl><dt>
          
          <code><a href="volcano_interactive.html">volcano_interactive()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="volcano_interactive_comparison.html">volcano_interactive_comparison()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="volcano_plot_deseq.html">volcano_plot_deseq()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="volcano_plot_limma.html">volcano_plot_limma()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl></div>
  </main></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Eléonore Schneegans, Nurun Fancy.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

